digest/tests/swap_digest.rs

```diff
@@ -0,0 +1,165 @@
+use digest::{digest, FieldDigest, RootDigest};
+use digest_macro_derive::DigestRootMacro;
+
+use digest::multihash::Multihash;
+
+struct NewType(String);
+
+impl RootDigest for NewType {
+    fn digest_root(self) -> Multihash {
+        self.0.digest_field("".into())
```

Comment:
For tuple structs, we should take into account the position, similarly how we use the field name for regular structs.

Reply:
Do we want to support tuples and new types? Maybe the best is to just remove this.

Reply:
Actually I changed it so I think it makes sense now.